src/vs/workbench/contrib/chat/browser/chatAttachmentResolveService.ts

```diff
@@ -198,24 +204,71 @@ export class ChatAttachmentResolveService implements IChatAttachmentResolveServi
 			resource: resource,
 			mimeType: mimeType,
 			omittedState: isPartiallyOmitted ? OmittedState.Partial : OmittedState.NotOmitted
-		}]);
+		}], updateCallback);
 
 		return imageFileContext[0];
 	}
 
-	public resolveImageAttachContext(images: ImageTransferData[]): Promise<IChatRequestVariableEntry[]> {
-		return Promise.all(images.map(async image => ({
-			id: image.id || await imageToHash(image.data),
-			name: image.name,
-			fullName: image.resource ? image.resource.path : undefined,
-			value: await resizeImage(image.data, image.mimeType),
-			icon: image.icon,
-			kind: 'image',
-			isFile: false,
-			isDirectory: false,
-			omittedState: image.omittedState || OmittedState.NotOmitted,
-			references: image.resource ? [{ reference: image.resource, kind: 'reference' }] : []
-		})));
+	public resolveImageAttachContext(images: ImageTransferData[], updateCallback?: (updatedEntry: IChatRequestVariableEntry) => void): Promise<IChatRequestVariableEntry[]> {
+		return Promise.all(images.map(async image => {
+			const binaryData = await resizeImage(image.data, image.mimeType);
+			const id = image.id || await imageToHash(binaryData);
+
+			const chatProviderId = this.productService.defaultChatAgent?.provider?.default?.id ?? '';
```

Comment:
I think you should handle undefined instead of defaulting to empty string since there is no such thing as a provider with the id `''`